biodatasets/bio_simlex/bio_simlex.py

```diff
@@ -0,0 +1,181 @@
+    # https://huggingface.co/docs/datasets/add_dataset.html#downloading-data-files-and-organizing-splits
+    # ds_source = datasets.load_dataset('my_dataset', name='source', data_dir="/path/to/data/files")
+    # ds_bigbio = datasets.load_dataset('my_dataset', name='bigbio', data_dir="/path/to/data/files")
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="bio_simlex_source",
+            version=SOURCE_VERSION,
+            description="bio_simlex source schema",
+            schema="source",
+            subset_id="bio_simlex",
+        ),
+        BigBioConfig(
+            name="bio_simlex_bigbio_pairs",
+            version=BIGBIO_VERSION,
+            description="bio_simlex BigBio schema",
+            schema="bigbio_pairs",
+            subset_id="bio_simlex",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "bio_simlex_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "text_1": datasets.Value("string"),
+                    "text_2": datasets.Value("string"),
+                    "score": datasets.Value("float"),
+                }
+            )
+
+        elif self.config.schema == "bigbio_pairs":
+            features = schemas.pairs_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        url = _URLS[_DATASETNAME]
+        data_dir = dl_manager.download_and_extract(url)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # Whatever you put in gen_kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": data_dir,
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        with open(filepath, "r", encoding="utf-8") as f:
+            for id_, line in enumerate(f):
+                word1, word2, score = line.split("\t")
+                if self.config.schema == "source":
+                    yield id_, {
+                        "text_1": word1,
+                        "text_2": word2,
+                        "score": score,
+                    }
+
+                elif self.config.schema == "bigbio_pairs":
+                    yield id_, {
+                        "id": str(id_),
+                        "document_id": "NULL",
+                        "text_1": word1,
+                        "text_2": word2,
+                        "label": "0" if float(score) < 5 else "1",
```

Comment:
discussed with @debajyotidatta and @jason-fries , for now I think you can leave the "label" as the string representation of the number without thresholding. can you also put a comment in about what the range is please (assuming 0-9?)

Reply:
done